instrumentation/github.com/aws/aws-lambda-go/otellambda/lambda.go

```diff
@@ -0,0 +1,158 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otellambda
+
+import (
+	"context"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/lambdacontext"
+
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
+)
+
+var errorLogger = log.New(log.Writer(), "OTel Lambda Error: ", 0)
+
+type Flusher interface {
+	ForceFlush(context.Context) error
+}
+
+type noopFlusher struct{}
+
+func (*noopFlusher) ForceFlush(context.Context) error { return nil }
+
+// Compile time check our noopFlusher implements FLusher
+var _ Flusher = &noopFlusher{}
+
+type InstrumentationOption func(o *InstrumentationOptions)
+
+type InstrumentationOptions struct {
```

Comment:
Please see [the contributing guidelines related to configuration](https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#configuration).  The struct should probably be called simply `config` and not exported.  The option type should be an interface and acan be called `Option` since it will be scoped with the package name if it is used elsewhere.

```suggestion
type Option interface {
	apply(*config)
}

type config struct {
```

A helper `Option` implementation can be added for options that are simple:

```go
type optionFunc func(*config)

func (o optionFunc) apply(c *config) {
	o(c)
}
```